circle.yml

```diff
@@ -4,58 +4,116 @@
+    - run: mkdir -p ~/downloads ~/go-path
+    - run: ls "$HOME/downloads"
+    - run: ls "$ANDROID_HOME"
+
+    - run: if [[ ! -e "$HOME/downloads/go.$GOVERSION.tar.gz" ]]; then wget -nv https://storage.googleapis.com/golang/go$GOVERSION.linux-amd64.tar.gz -O $HOME/downloads/go.$GOVERSION.tar.gz ; fi
+    - run: cd $HOME && tar -xzvf downloads/go.$GOVERSION.tar.gz
+    - run: sudo mv $HOME/go /usr/local
+
+    - run: if ! $(grep -q "Revision=24.4.1" $ANDROID_HOME/tools/source.properties); then echo y | android update sdk -u -a -t "tools"; fi
+    - run: if [ ! -e $ANDROID_HOME/build-tools/23.0.2 ]; then echo y | android update sdk -u -a -t "build-tools-23.0.2"; fi
+
+    # Android Support Repository, revision 35 / Local Maven repository for Support Libraries
+    - run: if [ ! -d "/usr/local/android-sdk-linux/extras/android/m2repository/com/android/support/design/24.1.0" ]; then echo y | android update sdk --no-ui --all --filter "extra-android-m2repository"; fi
+
     # Accept android licenses
-    - mkdir /usr/local/android-sdk-linux/licenses
-    - if [[ ! -e "$ANDROID_HOME/ndk_bundle" ]]; then wget -nv http://dl.google.com/android/repository/android-ndk-r11c-linux-x86_64.zip -O android-ndk-r11c-linux-x86_64.zip && unzip -o android-ndk-r11c-linux-x86_64.zip -d "$ANDROID_HOME" && mv "$ANDROID_HOME/android-ndk-r11c" "$ANDROID_HOME/ndk-bundle"; fi
-    - echo "8933bad161af4178b1185d1a37fbf41ea5269c55" > /usr/local/android-sdk-linux/licenses/android-sdk-license
-    # Install Yarn
-    - sudo apt-key adv --fetch-keys http://dl.yarnpkg.com/debian/pubkey.gpg
-    - echo "deb http://dl.yarnpkg.com/debian/ stable main" | sudo tee /etc/apt/sources.list.d/yarn.list
-    - sudo apt-get update -qq
-    - sudo apt-get install -y -qq yarn=1.0.2-1
-  cache_directories:
-    - /usr/local/android-sdk-linux/tools
-    - /usr/local/android-sdk-linux/build-tools/23.0.2
-    # - android-ndk-r11c
-  override:
-    - yarn install --pure-lockfile
-    - yarn global add react-native-cli
-    - ln -s $HOME/kbfs $GOPATH/src/github.com/keybase/kbfs
-    - (cd $GOPATH/src/github.com/keybase/client/shared; LOCAL_KBFS=1 yarn run rn-gobuild-android)
-    - ls -la $GOPATH/src/github.com/keybase/client/shared/react-native/android/keybaselib/keybaselib.aar
-test:
-  override:
-    # Just testing builds
-    - echo "Only testing builds"
+    - run: mkdir -p /usr/local/android-sdk-linux/licenses
+    - run: if [[ ! -e "$ANDROID_HOME/ndk-bundle" ]]; then wget -nv http://dl.google.com/android/repository/android-ndk-r11c-linux-x86_64.zip -O android-ndk-r11c-linux-x86_64.zip && unzip -o android-ndk-r11c-linux-x86_64.zip -d "$ANDROID_HOME" && mv "$ANDROID_HOME/android-ndk-r11c" "$ANDROID_HOME/ndk-bundle"; fi
+    - run: echo "8933bad161af4178b1185d1a37fbf41ea5269c55" > /usr/local/android-sdk-linux/licenses/android-sdk-license
+    - run: echo "d56f5187479451eabf01fb78af6dfcb131a6481e" >> /usr/local/android-sdk-linux/licenses/android-sdk-license
+
+    # Clone client and install Yarn
```

Comment:
This is the point where we diverge from https://github.com/keybase/client/pull/12371

Reply:
Seems to me like except for the checkout, everything up to here should be cached between builds.

Reply:
Hmm...I don't know if adding more caching would help. Looking at this recent run https://circleci.com/gh/keybase/kbfs/6116 , the slowest tasks are:

- Saving the cache (!) at ~5 minutes, since it looks like the cache is 2.1 GB.
- `yarn run rn-gobuild-android` at ~3:30
- Restoring the cache at ~1:10

then everything else at <10s or so.

Reply:
OK fair enough. Maybe saving the cache costs more than just redoing the work that needs to be cached in the first place?